### PR TITLE
zplugin:init at 2.3

### PR DESCRIPTION
**Pull request by pasqui23**
###### Things done

Building fails with 
```
nix-env -f ./. -iA zplugin
installing 'zplugin-2.3'
these derivations will be built:
  /nix/store/sanj1x0ishjzrqbpkwhpmwr41hl2j6vn-zplugin-2.3.drv
building '/nix/store/sanj1x0ishjzrqbpkwhpmwr41hl2j6vn-zplugin-2.3.drv'...
unpacking sources
unpacking source archive /nix/store/wl0pvhglyvkyvny2hgjvgyn0bi89ilnr-source
source root is source
patching sources
configuring
no configure script, doing nothing
installing
/nix/store/npsin1jlnzs5sqxpsv2mk5gbnm4hk4kr-stdenv-linux/setup: eval: line 1337: unexpected EOF while looking for matching `"'
builder for '/nix/store/sanj1x0ishjzrqbpkwhpmwr41hl2j6vn-zplugin-2.3.drv' failed with exit code 2
error: build of '/nix/store/sanj1x0ishjzrqbpkwhpmwr41hl2j6vn-zplugin-2.3.drv' failed
```

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/shells/zsh/zplugin/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchFromGitHub {
+    owner = "zdharma";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0qqv5p19s8jb06d6h55dm4acji9x2rpxb2ni3h7fb0q43iz6y85w";
+  };
+  # adapted from https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zsh-zplugin-git
+  dontBuild = true;
+  installPhase = ''
+    outdir="$out/share/$pname
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]:
```suggestion
    outdir="$out/share/$pname"
```

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

File: pkgs/shells/zsh/zplugin/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    # Zplugin-module files
+    find zmodules/ -type d -exec install -dm 755 "{}" "$outdir/{}" \;
+    find zmodules/ -type f -exec install -m 744 "{}" "$outdir/{}" \;
+
+  '';
+  #TODO:doc output
+  
+  meta = with lib; {
+    license = licenses.mit;
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]: license is listed twice 
```suggestion
```

File: pkgs/shells/zsh/zplugin/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    homepage = "https://github.com/zdharma/zplugin";
+    description = "Flexible zsh plugin manager";
+    license = licenses.mit;
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]: please add yourself as a maintainer :)
```suggestion
    license = licenses.mit;
    maintainers = with maintainers; [ pasqui23 ];
```

Comment (pasqui23) [reply pkgs/shells/zsh/zplugin/default.nix]: Wait I'm not listed among the mantainers,can I add myself?

**Comment by pasqui23** [comment]
What is the exact procedure for proposing as mantainer?

**Comment by jonringer** [comment]
it's pretty relaxed in nixpkgs, the main qualification is just general interest in the package. Most of the time, maintainers know how to use the software so they can verify that updates to it don't break any workflows.

**Comment by little-dude** [comment]
@pasqui23 do you plan to push this forward? Otherwise I might try to pick it up.

**Comment by pasqui23** [comment]
So apparently it builds with the latest code

**Comment by pasqui23** [comment]
@little-dude yes,I still want to push it

File: pkgs/shells/zsh/zplugin/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{ stdenvNoCC, lib, fetchFromGitHub }:
+stdenvNoCC.mkDerivation rec {
+  pname = "zplugin";
+  version = "2.3";
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]:
```suggestion
  version = "2.3";

```

File: pkgs/shells/zsh/zplugin/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{ stdenvNoCC, lib, fetchFromGitHub }:
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]:
```suggestion
{ stdenvNoCC, lib, fetchFromGitHub }:

```

File: pkgs/shells/zsh/zplugin/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    cd "$src"
+
+    # Zplugin's source files
+    install -dm0755 "$outdir"
+    install -m0644 zplugin{,-side,-install,-autoload}.zsh "$outdir"
+    install -m0755 git-process-output.zsh "$outdir"
+
+    # Zplugin autocompletion
+    install -dm0755 "$out/share/zsh/site-functions"
+    install -m0744 _zplugin "$out/share/zsh/site-functions"
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]: can be replaced with 
```suggestion
    installShellCompletion --zsh _zplugin
```

File: pkgs/shells/zsh/zplugin/default.nix [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+    cd "$src"
+
+    # Zplugin's source files
+    install -dm0755 "$outdir"
+    install -m0644 zplugin{,-side,-install,-autoload}.zsh "$outdir"
+    install -m0755 git-process-output.zsh "$outdir"
+
+    # Zplugin autocompletion
+    install -dm0755 "$out/share/zsh/site-functions"
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]: unnecessary with other suggestion
```suggestion
```

**Review by jonringer — COMMENTED** [review]
you could also use `install -DmXXX [sources] -t dest` to greatly simplify a lot of this installation logic (it will create parent directories for you)

File: pkgs/shells/zsh/zplugin/default.nix [hunk 8]
```diff
@@ -0,0 +1,39 @@
+{ stdenvNoCC, lib, fetchFromGitHub }:
+stdenvNoCC.mkDerivation rec {
+  pname = "zplugin";
+  version = "2.3";
+  src = fetchFromGitHub {
+    owner = "zdharma";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0qqv5p19s8jb06d6h55dm4acji9x2rpxb2ni3h7fb0q43iz6y85w";
+  };
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]:
```suggestion
  };
  
  nativeBuildInputs = [ installShellCompletion ];
```

File: pkgs/shells/zsh/zplugin/default.nix [hunk 9]
```diff
@@ -0,0 +1,39 @@
+{ stdenvNoCC, lib, fetchFromGitHub }:
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]:
```suggestion
{ stdenvNoCC, lib, fetchFromGitHub, installShellCompletion }:
```

**Review by jonringer — COMMENTED** [review]
sorry, forgot to add that installShellCompletion is another "package", so it needs to be added to avoid:

```
/nix/store/plcn9z5lzjd9lhij1cj73k2nwxkng4rq-stdenv-linux/setup: line 1309: installShellCompletion: command not found
```

**Comment by jonringer** [comment]
don't forget to squash the commits, to squash the last 6 commits into the first you can do:

```
git reset HEAD~5 # put the last 5 changes into unstaged
git add pkgs/
git commit --amend --no-edit # amend into previous commit
git push <fork> <branch> --force
```

for other incremental changes into the last commit, you can just do:
```
git add pkgs/
git commit --amend --no-edit # amend into previous commit
git push <fork> <branch> --force
```

File: pkgs/shells/zsh/zplugin/default.nix [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    install -dm0755 "$outdir"
+    install -m0644 zplugin{,-side,-install,-autoload}.zsh "$outdir"
+    install -m0755 git-process-output.zsh "$outdir"
+
+    # Zplugin autocompletion
+    installShellCompletion --zsh _zplugin
+
+    # Zplugin-module files
+    find zmodules/ -type d -exec install -dm 755 "{}" "$outdir/{}" \;
+    find zmodules/ -type f -exec install -m 744 "{}" "$outdir/{}" \;
```
Comment (jonringer) [inline pkgs/shells/zsh/zplugin/default.nix]: should this preserve the same file structure?
```suggestion
    find zmodules/ -type f -exec install -m 744 "{}" "$outdir/zmodules/{}" \;
```

Comment (pasqui23) [reply pkgs/shells/zsh/zplugin/default.nix]: Honestly now that I look it better,it seem it would be better if packaged separately

**Review by wmertens — COMMENTED** [review]
Triage: lgtm but wondering about the native inputs @jonringer mentioned, should they be merged with the one on line 13?

File: maintainers/maintainer-list.nix [hunk 11]
```diff
@@ -5128,6 +5128,11 @@
     githubId = 131844;
     name = "Igor Pashev";
   };
+  pasqui23 = {
+    email = "p3dimaria@hotmail.it";
+    github = "pasqui23";
+    githubId = 6931743;
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:5135]: Please add your name.

Please make this into a separate commit:

```
maintainers: add <name>
```

**Review by jonringer — COMMENTED** [review]
please resolve conflicts as well :)

**Comment by pasqui23** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/90322 @jonringer
